src/main/java/by/custom/utilcalculator/domain/FileUserProgressStorage.java

```diff
@@ -0,0 +1,60 @@
+package by.custom.utilcalculator.domain;
+
+import by.custom.utilcalculator.exception.UserFileNotFoundException;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class FileUserProgressStorage implements iUserProgressStorage {
+    private final String path = "D:/%s.dat";
+
+    private static class FileUserProgressStorageHolder {
+        private static final FileUserProgressStorage FILE_USER_PROGRESS_STORAGE = new FileUserProgressStorage();
+    }
+
+    private FileUserProgressStorage() {
+    }
+
+    public static FileUserProgressStorage getInstance() {
+        return FileUserProgressStorageHolder.FILE_USER_PROGRESS_STORAGE;
+    }
+
+    @Override
+    public void save(UserProgress userProgress) {
+        try {
+            ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(String.format(path, userProgress.getChatID())));
+            oos.writeObject(userProgress);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public UserProgress get(String chatID) throws UserFileNotFoundException {
+        UserProgress localUserProgress;
+        if (isUserFileExists(chatID)) {
```

Comment:
To avoid redundant inner block there is "fail-fast" approach:
```suggestion
        if (!isUserFileExists(chatID)) {
            throw new UserFileNotFoundException();
        }
        try {
                ObjectInputStream ois = new ObjectInputStream(new FileInputStream(String.format(path, chatID)));
               return (UserProgress) ois.readObject();
            } catch (IOException | ClassNotFoundException e) {
                throw new RuntimeException(e);
            }
```

Reply:
Done.